src/lib/engine/guesstimate.js

```diff
@@ -82,3 +82,15 @@ function _inputMetricsWithValues(guesstimate: Guesstimate, dGraph: DGraph): Obje
   })
   return [inputs, _.uniq(errors)]
 }
+
+export function inputsFromExpressions(metrics) {
+  let idMap = {}, reParts = []
+  metrics.forEach( ({id, readableId}) => {reParts.push(`\\\$\\\{${id}\\\}`); idMap[`\$\{${id}\}`] = readableId} )
```

Comment:
Maybe make functions that return these regex parts and name them on the top of this file?  The idMap is used more than once.  Comments with these would help.